pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/CompositeTransformer.java

```diff
@@ -41,6 +41,32 @@
 public class CompositeTransformer implements RecordTransformer {
   private final List<RecordTransformer> _transformers;
 
+  /**
+   * Returns a list of record transformers that perform enrichment of the record before the record is passed onto the
+   * ComplexType transformer. The transform pipeline order is as follows: pre-enrichers, complex type transformers,
+   * other record transformers.
+   */
+  public static List<RecordTransformer> getDefaultPreEnrichers(TableConfig tableConfig, Schema schema) {
```

Comment:
Do you also need to modify `getDefaultTransformers()` to exclude the pre enrichers?
Suggest pulling out the common part